mbed-client/m2mbase.h

```diff
@@ -243,13 +243,13 @@ class M2MBase : public M2MReportObserver {
      * \brief Returns the interface description of the object.
      * \return Description of the object.
      */
-    virtual const String& interface_description() const;
+    String interface_description() const;
```

Comment:
Is the virtual removal done on purpose? Is this or other method which lost the virtual overridden in child classes?